agent/consul/server_test.go

```diff
@@ -1119,3 +1121,24 @@ func TestServer_RPC_RateLimit(t *testing.T) {
 		}
 	})
 }
+
+func TestServer_CALogging(t *testing.T) {
+	t.Parallel()
+	dir1, conf1 := testServerConfig(t)
+	s1, err := NewServer(conf1)
+	if err != nil {
+		t.Fatalf("err: %v", err)
+	}
+	defer os.RemoveAll(dir1)
+	defer s1.Shutdown()
+	testrpc.WaitForLeader(t, s1.RPC, "dc1")
+
+	if _, ok := s1.caProvider.(ca.NeedsLogger); !ok {
+		t.Fatalf("provider does not implement NeedsLogger")
+	}
+
+	v := reflect.ValueOf(s1.caProvider).Elem().FieldByName("logger")
```

Comment:
Is using `reflect` an appropriate way to check this? Is the intent to check that for a CA provider that implements `NeedsLogger`, the `logger` field gets set? Is there a need for testing the inverse (i.e. `logger` is undefined if `NeedsLogger` is not implemented?)